src/app/exporting/exporting.component.ts

```diff
@@ -0,0 +1,148 @@
+import { Component, OnInit, Input, ViewEncapsulation } from '@angular/core';
+
+@Component({
+  selector: 'exporting',
+  templateUrl: './exporting.component.html',
+  styleUrls: ['./exporting.component.less'],
+  encapsulation: ViewEncapsulation.None
+})
+export class ExportingComponent implements OnInit {
+  @Input() jsonData: any;
+
+  exportOpen = false;
+  jsonContainer;
+  htmlContainer;
+  jsonString: string;
+  timeNow = new Date().toLocaleDateString();
+
+  constructor() { }
+
+  ngOnInit() {
+    this.jsonContainer = document.getElementById('json-container');
+    this.htmlContainer = document.getElementById('html-container');
+  }
+
+  clearExports() {
+    while (this.htmlContainer.firstChild) {
+      this.htmlContainer.removeChild(this.htmlContainer.firstChild);
+    }
+    while (this.jsonContainer.firstChild) {
+      this.jsonContainer.removeChild(this.jsonContainer.firstChild);
+    }
+  }
+
+  exportData() {
```

Comment:
You could extract this logic out to an export service as well, if you wanted to clean up the component.